### PR TITLE
Add a README

### PR DESCRIPTION
**Pull request by garetht**
Adds a readme to the repository!

- Summary of the work, with original paper diagram
- Our modifications
- Results
- Diagram of the RL flow
- Structure of the repository
- Future work!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,138 @@
...
+
+The abduction and induction tasks are determined in the same way, where the solver has to solve for $y$ and $p$ respectively. Like the induction task in the paper, multiple primes are possible, and we will accept any of those that fulfill the task for a given $x$ and $y$.
+
+We can control the computational difficulty of the problem by increasing the size of possible $p$ candidates. This makes this problem more amenable to RL since we are able to tune the problem difficulty to match the initial capability of the model.
+
+## Setup
+
+We followed the Absolute Zero paper in training on a model in the `Qwen/Qwen2.5-3B` series, but we started with `Qwen/Qwen2.5-3B-Instruct` instead of a base model, because we found it unproductive to attempt to instruct the base model to respond in a parseable format for us to correctly reward it and bootstrap its learning.
+
+We ran this training on Nvidia A100s in a cloud-provisioned instance on RunPod.
```
Comment (pragmaticTNT) [inline README.md]: Didn't we run it on eight A100s?

File: README.md [hunk 2]
```diff
@@ -1 +1,138 @@
...
+
+## Results
+
+Over 200 RL training steps, we saw sharp increases in the capability of our proposer for all three problem types – abduction, induction and deduction – over the last twenty steps, as expected.
```
Comment (pragmaticTNT) [inline README.md]: *Towards the end of* "over 200 RL training steps..."

Comment (pragmaticTNT) [reply README.md]: Why did we expect this to happen in the last twenty steps?

File: README.md [hunk 3]
```diff
@@ -1 +1,138 @@
...
+
+The induction task from the paper becomes a task to solve for $x$, e.g. 
+
+$$x \times 4 \equiv 1 \pmod{7}$$
+
+and we can determine that, uniquely, $x \equiv 2 \pmod{7}$ in this example.
+
+The abduction and induction tasks are determined in the same way, where the solver has to solve for $y$ and $p$ respectively. Like the induction task in the paper, multiple primes are possible, and we will accept any of those that fulfill the task for a given $x$ and $y$.
+
+We can control the computational difficulty of the problem by increasing the size of possible $p$ candidates. This makes this problem more amenable to RL since we are able to tune the problem difficulty to match the initial capability of the model.
```
Comment (pragmaticTNT) [inline README.md]: This makes *the* problem...

File: README.md [hunk 4]
```diff
@@ -1 +1,138 @@
...
+    B9 --- D2[compute_r_total]
+```
+
+## Repository Layout
+
+- `custom_types.py` - Core data structures and type definitions used throughout the codebase, including representations for problems, answers, batches, and enums for task types and roles.
+
+- `constants.py` - Global constants such as model names, directory paths, device selection logic, and problem parameters (e.g., maximum prime).
+
+- `prompts.py` - Prompt templates and prompt-generation utilities, including those adapted from the original Absolute Zero paper, for constructing tasks and evaluating model outputs.
```
Comment (pragmaticTNT) [inline README.md]: I'm pretty sure this file is never used and can be deleted

File: README.md (hunk 2 above)
Comment (wingardl) [inline README.md]: over last 20 or over 200? wording is a bit off

**Review by wingardl — APPROVED** [review]

**Review by pragmaticTNT — APPROVED** [review]
Looks good to me!

**Review by wingardl — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -1 +1,136 @@
...
+
+The abduction and induction tasks are determined in the same way, where the solver has to solve for $y$ and $p$ respectively. Like the induction task in the paper, multiple primes are possible, and we will accept any of those that fulfill the task for a given $x$ and $y$.
+
+We can control the computational difficulty of the problem by increasing the size of possible $p$ candidates. This makes the problem more amenable to RL since we are able to tune the problem difficulty to match the initial capability of the model.
+
+## Setup
+
+We followed the Absolute Zero paper in training on a model in the `Qwen/Qwen2.5-3B` series, but we started with `Qwen/Qwen2.5-3B-Instruct` instead of a base model, because we found it unproductive to attempt to instruct the base model to respond in a parseable format for us to correctly reward it and bootstrap its learning.
+
+We ran this training on a cluster of eight Nvidia A100s in a cloud-provisioned instance on RunPod.
```
Comment (wingardl) [inline README.md]: our final run was actually just 1 A100, we never got the 8 to work

**Review by wingardl — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (pragmaticTNT) [reply README.md]: Oops, ok lemme fix that

**Review by wingardl — APPROVED** [review]
